target/linux/pistachio/base-files/lib/functions/mac.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+#
+# Copyright 2016 by Imagination Technologies Limited and/or its affiliated group companies.
+#
+# All rights reserved.  No part of this software, either
+# material or conceptual may be copied or distributed,
+# transmitted, transcribed, stored in a retrieval system
+# or translated into any human or computer language in any
+# form by any means, electronic, mechanical, manual or
+# other-wise, or disclosed to the third parties without the
+# express written permission of Imagination Technologies
+# Limited, Home Park Estate, Kings Langley, Hertfordshire,
+# WD4 8LZ, U.K.
+#
+# common shell lib to obtain wifi mac address
+
+
+get_wifi_macaddr() {
+	local wifi_mac_addr=0x$(hexdump -n 6 -v -e '/1 "%02X"' /proc/device-tree/uccp@18480000/mac-address0);
+	if [ $? -ne 0 ]
+	then
+		echo -e "FAIL" >/dev/null
+		exit 1;
```

Comment:
Using a mix of exit codes and string returns makes an unclear API. Perhaps instead just return empty string and expect that to be tested for by caller?

Whichever way you go with if something goes wrong then their should be a loud WARNING left in logs about it.